indico/modules/events/registration/api.py

```diff
@@ -37,74 +38,53 @@ class RHAPIRegistrant(RH):
     """RESTful registrant API"""
 
     @oauth.require_oauth('registrants')
+    def _checkProtection(self):
+        if not self.event.can_manage(request.oauth.user, role='registration'):
+            raise Forbidden()
+
     def _checkParams(self):
-        event = ConferenceHolder().getById(request.view_args['event_id'], True)
-        if event is None:
+        self.event = Event.find(id=request.view_args['event_id'], is_deleted=False).first_or_404()
+        if self.event is None:
             raise NotFound("No such event")
-        if not event.canManageRegistration(request.oauth.user):
-            raise Forbidden()
-        registrant = event.getRegistrantById(request.view_args['registrant_id'])
-        if registrant is None:
-            raise NotFound("No such registrant")
-        self.event = event
-        self.registrant = registrant
-
-    def _get_result(self, **kwargs):
-        checkin_date = None
-        if self.registrant.isCheckedIn():
-            checkin_date = self.registrant.getCheckInDate().isoformat()
-        return jsonify(event_id=self.event.getId(),
-                       registrant_id=self.registrant.getId(),
-                       full_name=self.registrant.getFullName(title=True, firstNameFirst=True),
-                       checked_in=self.registrant.isCheckedIn(),
-                       checkin_secret=self.registrant.getCheckInUUID(),
-                       checkin_date=checkin_date,
-                       registration_date=self.registrant.getRegistrationDate().isoformat(),
-                       **kwargs)
+
+        self._registration = (self.event.registrations
```

Comment:
i wouldn't underscore it (in new code we rarely underscore things that are used a lot like self.event etc)